BGL/include/CGAL/boost/graph/helpers.h

```diff
@@ -1123,15 +1123,18 @@ clear_impl(FaceGraph& g)
   typedef typename boost::graph_traits<FaceGraph>::edge_descriptor     edge_descriptor;
   typedef typename boost::graph_traits<FaceGraph>::vertex_descriptor   vertex_descriptor;
   typedef typename boost::graph_traits<FaceGraph>::face_descriptor     face_descriptor;
-  BOOST_FOREACH(edge_descriptor ed, edges(g)) {
+
+  std::vector<edge_descriptor> edges_to_rm(edges(g).first, edges(g).second);
+  BOOST_FOREACH(edge_descriptor ed, edges_to_rm)
     remove_edge(ed, g);
-  }
-  BOOST_FOREACH(vertex_descriptor vd, vertices(g)) {
+
+  std::vector<vertex_descriptor> vertices_to_rm(vertices(g).first, vertices(g).second);
+  BOOST_FOREACH(vertex_descriptor vd, vertices_to_rm)
     remove_vertex(vd, g);
-  }
-  BOOST_FOREACH(face_descriptor fd, faces(g)) {
+
+  std::vector<face_descriptor> faces_to_rm(faces(g).first, faces(g).second);
+  BOOST_FOREACH(face_descriptor fd, faces_to_rm)
     remove_face(fd, g);
-  }
 }
```

Comment:
I think that this implementation is as wrong as the previous one, because a face graph could have its descriptors invalidated by `remove_*. For example the `CGAL::Polyhedron_3` with a vector as container.

Reply:
It should not otherwise almost nothing would work.

Reply:
This means that the polyhedron based on vector is not a valid model.

Reply:
However I agree that nothing is say in the boost [concept](https://www.boost.org/doc/libs/1_67_0/libs/graph/doc/MutableGraph.html)

Reply:
[`adjacency_list`] invalidates its descriptors in `remove_vertex`, when the `VertexList` template parameter is `vecS`. See the section "Iterator and Descriptor Stability/Invalidation" in the documentation.

[`adjacency_list`]: https://www.boost.org/doc/libs/1_66_0/libs/graph/doc/adjacency_list.html

Reply:
I continue to think the same thing but I've changed the patch since it was making it simpler :)